### PR TITLE
Convert Application to a functional component, convert to TypeScript

### PR DESCRIPTION
**Pull request by martinpitt**
- [x] Needs https://github.com/cockpit-project/cockpit/pull/20745
 - [x] https://github.com/cockpit-project/starter-kit/pull/918

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,7 +46,8 @@
     "stylelint-config-recommended-scss": "14.0.0",
     "stylelint-config-standard": "36.0.1",
     "stylelint-config-standard-scss": "13.1.0",
-    "stylelint-formatter-pretty": "4.0.0"
+    "stylelint-formatter-pretty": "4.0.0",
+    "typescript": "^5.5.3"
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply package.json]: Thanks for spotting! Fixed.

**Review by allisonkarlitskaya — APPROVED** [review]
Done and dusted.  :broom:

File: src/app.tsx [hunk 2]
```diff
@@ -17,34 +17,32 @@
...
 
-        cockpit.file('/etc/hostname').watch(content => {
-            this.setState({ hostname: content.trim() });
-        });
-    }
+    useEffect(() => {
+        const hostname = cockpit.file('/etc/hostname');
+        hostname.watch(content => setHostname(content?.trim() ?? ""));
+        return hostname.close;
+    });
```
Comment (subhoghoshX) [inline src/app.tsx:35]: This doesn't look right. I think it should have a empty dependency array to limit it to only run once after mount.

Comment (martinpitt) [reply src/app.tsx:35]: @subhoghoshX Oh, you are right! Fixed in #920.

... and nice to see you again! :wave:

Comment (subhoghoshX) [reply src/app.tsx:35]: Hey, haha I just opened a PR. Maybe I should close it.

**Pull request merged**